### PR TITLE
tests: add pmemspoil to cmake

### PR DESCRIPTION
**Pull request by PietrasMaciej**
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmdk-convert/47)

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -341,6 +341,52 @@ if(BUILD_CONVERT_V5)
...
+                ${COMMON_SOURCES}
+		)
+
+	if(WIN32)
+	add_executable(pmemspoil ${PMEMSPOIL_SOURCES} ${CMAKE_SOURCE_DIR}/nvml-1.5/src/windows/getopt/getopt.c)
+
+	target_include_directories(pmemspoil PRIVATE ${CMAKE_SOURCE_DIR}/nvml-1.5/src/windows/getopt)
+	target_include_directories(pmemspoil PRIVATE ${CMAKE_SOURCE_DIR}/nvml-1.5/src/windows/include)
+	else()
+	add_executable(pmemspoil ${PMEMSPOIL_SOURCES})
```
Comment (marcinslusarz) [inline tests/CMakeLists.txt]: You don't have to duplicate add_executable in both cases. Append getopt.c to PMEMSPOIL_SOURCES in WIN32 path and use single add_executable after.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -341,6 +341,52 @@ if(BUILD_CONVERT_V5)
...
+	set(PMEMSPOIL_SOURCES
+                ${CMAKE_SOURCE_DIR}/nvml-1.5/src/test/tools/pmemspoil/spoil.c
+                ${CMAKE_SOURCE_DIR}/nvml-1.5/src/tools/pmempool/common.c
+                ${CMAKE_SOURCE_DIR}/nvml-1.5/src/libpmemblk/btt.c
+                ${CMAKE_SOURCE_DIR}/nvml-1.5/src/tools/pmempool/output.c
+                ${COMMON_SOURCES}
+		)
+
+	if(WIN32)
+	add_executable(pmemspoil ${PMEMSPOIL_SOURCES} ${CMAKE_SOURCE_DIR}/nvml-1.5/src/windows/getopt/getopt.c)
```
Comment (marcinslusarz) [inline tests/CMakeLists.txt]: indentation

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -341,6 +341,52 @@ if(BUILD_CONVERT_V5)
...
+
+	target_include_directories(pmemspoil PRIVATE ${CMAKE_SOURCE_DIR}/nvml-1.5/src/windows/getopt)
+	target_include_directories(pmemspoil PRIVATE ${CMAKE_SOURCE_DIR}/nvml-1.5/src/windows/include)
+	else()
+	add_executable(pmemspoil ${PMEMSPOIL_SOURCES})
+	endif()
+
+	if(MSVC)
+	target_include_directories(pmemspoil PRIVATE ${CMAKE_SOURCE_DIR}/include/)
```
Comment (marcinslusarz) [inline tests/CMakeLists.txt]: indentation

**Comment by marcinslusarz** [comment]
One more thing - in one PR fixes to previous commits in the same PR should be squashed to those commits.

**Comment by PietrasMaciej** [comment]
Ok, so if I understood you correctly, I have squashed to one commit this with fixes.

**Comment by marcinslusarz** [comment]
Yes, that's what I meant.

**Pull request merged**